airflow/models.py

```diff
@@ -4037,11 +4076,27 @@ def get_many(
 
         query = (
             session.query(cls)
-            .filter(and_(*filters))
-            .order_by(cls.execution_date.desc(), cls.timestamp.desc())
-            .limit(limit))
-
-        return query.all()
+                .filter(and_(*filters))
+                .order_by(cls.execution_date.desc(), cls.timestamp.desc())
+                .limit(limit))
+        results = query.all()
+        if results:
+            if enable_pickling is None:
+                enable_pickling = configuration.getboolean('core', 'enable_xcom_pickling')
+            for result in results:
+                if enable_pickling:
+                    result.value = pickle.loads(result.value)
+                else:
+                    try:
+                        result.value = json.loads(result.value.decode('UTF-8'))
+                    except ValueError:
+                        logging.error("Could not parse the XCOM value as JSON. "
+                                      "If you are using pickles instead of JSON "
+                                      "for XCOM, then you need to enable pickle "
+                                      "support for XCOM in your airflow config.")
+                        raise
+            return results
+        return None
```

Comment:
`return None` at the end of a function doesn't do anything. It might make more sense to return `[]` here